### PR TITLE
Transfer_learning_Karina

### PR DESCRIPTION
**Pull request by Karina9510**

### CODE REVIEW

**Comment by Karina9510** [comment]
V tom to pull requestu jsou dva issues #59 a #65 
Kdyz jsem pushovala dalsi ulohu #65, ona se mi automaticky pridala uz do meho existujiciho pull requestu. 
Nepodarilo se mi to rozdelit

**Comment by Karina9510** [comment]
Comment  split_and_display_channels_of_color_image_Karina jsem opravila na split_and_display_channels_of_color_image_Karina_corrected

File: input/posts/en/transfer_learning.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    layersToStay
+    fullyConnectedLayer(3) %creates a fully connected layer with an output size of 3
+    softmaxLayer %creates a softmax layer
+    classificationLayer %creates a classification layer
+    ]
+
+options = trainingOptions('sgdm',... %set training options: the stochastic gradient with momentum optimizer
+    'Shuffle', 'every-epoch',... %shuffles the training data before each training epoch and...
+    							 %shuffles the validation data before each network validation
+    'Verbose', false, 'plots', 'training-progress',... % display training progress information in the command window
```
Comment (tesar-tech) [inline input/posts/en/transfer_learning.md]: this comment is not true. Its the absolute opposite.

File: input/posts/en/transfer_learning.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+﻿title: Transfer learning
+lead: Convolutional neural network training (CNN)
+Published: 2018-12-31
+Tags: [matlab, transfer learning]
+prerequisites: [Matlab]
```
Comment (tesar-tech) [inline input/posts/en/transfer_learning.md]: add alexnet matlab addon (see other posts with alexnet)

File: input/posts/en/transfer_learning.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+﻿title: Transfer learning
+lead: Convolutional neural network training (CNN)
```
Comment (tesar-tech) [inline input/posts/en/transfer_learning.md]: This should be brief text about "what is going on". Explain the whole post in (approx) two sentences. If you are able to explain it in this way that means you understand it well..

File: input/posts/en/transfer_learning.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
+﻿title: Transfer learning
```
Comment (tesar-tech) [inline input/posts/en/transfer_learning.md]: change title and file name to something more specific (information about shapes and alexnet should be included)

File: input/posts/en/transfer_learning.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+lead: Convolutional neural network training (CNN)
+Published: 2018-12-31
+Tags: [matlab, transfer learning]
+prerequisites: [Matlab]
+Authors: [tesar-tech, Karina9510]
+---
+This script carry out  a convolutional neural network training (CNN). The training of image classification problems is provided by the input image data stores, the network architecture layers, and the training options. Validation accuracy of the method is 100%.
```
Comment (tesar-tech) [inline input/posts/en/transfer_learning.md]: Too lengthy. Add information about created db with link to previous post. It has to make sense somehow... Also change information about accuracy from "100 %" to "almost 100 %". It depends on training data..

File: input/posts/en/transfer_learning.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+layers = [
+    layersToStay
+    fullyConnectedLayer(3) %creates a fully connected layer with an output size of 3
+    softmaxLayer %creates a softmax layer
+    classificationLayer %creates a classification layer
+    ]
+
+options = trainingOptions('sgdm',... %set training options: the stochastic gradient with momentum optimizer
+    'Shuffle', 'every-epoch',... %shuffles the training data before each training epoch and...
+    							 %shuffles the validation data before each network validation
```
Comment (tesar-tech) [inline input/posts/en/transfer_learning.md]: this line makes the code invalid

File: input/posts/en/transfer_learning.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+%subfolder of the comma-separated pair,label data and name
+
+[imdsTrain, imdsValidation] = splitEachLabel(imds, 0.7);
+%split imds to two datastores specifying that each datastore contains 70% of each label and ...
+%the corresponding files
+
+layersToStay = net.Layers(1:22) % creating new layers
+layers = [
+    layersToStay
+    fullyConnectedLayer(3) %creates a fully connected layer with an output size of 3
```
Comment (tesar-tech) [inline input/posts/en/transfer_learning.md]: add WeightLearnRateFactor and BiasLearnRateFactor parameters. Explain why.

File: input/posts/en/transfer_learning.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+net = alexnet;
+imds = imageDatastore('imgs_shapes', 'IncludeSubfolders', true, 'LabelSource', 'foldername') 
+%creates a datastore from the collection of image data specified by location,...
+%subfolder of the comma-separated pair,label data and name
+
+[imdsTrain, imdsValidation] = splitEachLabel(imds, 0.7);
+%split imds to two datastores specifying that each datastore contains 70% of each label and ...
+%the corresponding files
+
+layersToStay = net.Layers(1:22) % creating new layers
```
Comment (tesar-tech) [inline input/posts/en/transfer_learning.md]: This line is not about creating new layers..

File: input/posts/cs/bleached_channels_of_color_image.md [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+Pro implementaci jen jednoho barevného RGB kanáu na obrázku je nutno natavit hodnoty ostatních barev na nulu.
+
+``` matlab
+A = imread('kytka256.jpg');
+R = A; G = A; B = A; 
+R(:,:,[2,3])= 0;
+G(:,:,[1,3])=0;
+B(:,:,[1,2])=0;
+
+montage({insertText(A,[1 1],'puvodni','FontSize',22),...
```
Comment (tesar-tech) [inline input/posts/cs/bleached_channels_of_color_image.md]: jo, tohle je hezký

File: input/posts/cs/split_and_display_channels_of_color_image.md [hunk 10]
```diff
@@ -0,0 +1,22 @@
+﻿title: Jednotlivé barevné kanály
+lead: Rozdělení a zobrazení barevných kanálů obrázku
+Published: 2019-01-03
+Tags: [matlab, RGB, channels]
+datazooFiles: imgs/kytka256.jpg
+prerequisites: [Matlab]
+Authors: [tesar-tech, Karina9510]
+---
+Script zobrazuje jednotlivé barevné kanály na RGB obrázku.
```
Comment (tesar-tech) [inline input/posts/cs/split_and_display_channels_of_color_image.md]: Celý ten popis je hrozně zmatený. Zkus to jednoduše, nehledej v tom složité věci...

**Pull request closed without merging**